src/rail/cli/pipe_scripts.py

```diff
@@ -84,180 +138,129 @@ def inspect(config_file):
     return 0
 
 
-def truth_to_observed_pipeline(
-    project,
-    selection,
-    flavor,
-    run_mode=RunMode.bash,
-):
-    pipeline_name = "truth_to_observed"
-    pipeline_info = project.get_pipeline(pipeline_name)
-    pipeline_path = project.get_path('pipeline_path', pipeline=pipeline_name, flavor=flavor)
+class PipelineCatalogConfiguration:
+
+    def __init__(
+        self,
+        project: RailProject,
+        source_catalog_tag: str,
+        sink_catalog_tag: str,
+        source_catalog_basename: str | None=None,
+        sink_catalog_basename: str | None=None,
+    ):
+        self._project = project
+        self._source_catalog_tag = source_catalog_tag
+        self._sink_catalog_tag = sink_catalog_tag
+        self._source_catalog_basename = source_catalog_basename
+        self._sink_catalog_basename = sink_catalog_basename
+
+    def get_source_catalog(self, **kwargs: Any) -> str:
+        return self._project.get_catalog(
+            self._source_catalog_tag, basename=self._source_catalog_basename, **kwargs,
+        )
 
-    input_catalog_name = pipeline_info['InputCatalogTag']
-    input_catalog = project.get_catalogs().get(input_catalog_name)
+    def get_sink_catalog(self, **kwargs: Any) -> str:
+        return self._project.get_catalog(
+            self._sink_catalog_tag, basename=self._sink_catalog_basename, **kwargs,
+        )
 
-    # Loop through all possible combinations of the iteration variables that are
-    # relevant to this pipeline
-    if (iteration_vars := input_catalog.get("IterationVars")) is not None:
-        iterations = itertools.product(
-            *[
-                project.config.get("IterationVars").get(iteration_var)
-                for iteration_var in iteration_vars
-            ]
+    def get_script_path(self, pipeline_name: str, sink_dir: str, **kwargs: Any) -> str:
+        return os.path.join(
+            sink_dir,
+            "submit_{pipeline_name}_{selection}_{flavor}.sh".format(pipeline_name=pipeline_name, **kwargs),
```

Comment:
the previous version of this code guaranteed that `selection` and `flavor` were defined so that the format would work. this will fail if the user doesn't specify `selection` and `flavor` in the keyword arguments, so we should probably make those explicitily part of the function signature?